src/utils.ts

```diff
@@ -0,0 +1,8 @@
+export function valueIsEmpty(value: any) { 
+    if (value && value.constructor === Array){
+        return value.length === 0
+    } else if (typeof(value) === 'boolean') {
```

Comment:
As the role of this function is to check for empty values rather than do any validation, it might make more sense for zero values not to trigger it, such as:
`} else if (typeof(value) === 'boolean' || value === 0) {`
In the example form, 0 is not considered a valid value so I decided to make it trigger the red required text. If we want the required text to have more of a strict validation function, it will likely need to receive arguments from the related input form as to what constitutes a valid value.

Reply:
I'm pretty sure we do need to allow 0 values  - could be a perfectly valid value for a number input. Yeah, validation should be separate from required.